tests/fixtures/invalid_pyproject/pyproject.toml

```diff
@@ -17,3 +17,4 @@ classifiers = [
 python = "*"
 pendulum = {"version" = "^2.0.5", allows-prereleases = true}
 invalid = "1.0"
+invalid_source = { "version" = "*", source = "not-exists" }
```

Comment:
Maybe, you can add an example with a multiple-constraints dependency and improve the implementation of this check.

```
invalid_source_multi = [
    { "version" = "*", platform = "linux", source = "exists" },
    { "version" = "*", platform = "win32", source = "not-exists2" },
]
```